File: cpp-answers/231.2-的幂.cpp
/*
 * @Author: your name
 * @Date: 2020-06-09 08:40:24
 * @LastEditTime: 2020-06-09 08:40:54
 * @LastEditors: your name
 * @Description: In User Settings Edit
 * @FilePath: \undefinede:\leetcode刷题\231.2-的幂.cpp
 */
/*
 * @lc app=leetcode.cn id=231 lang=cpp
 *
 * [231] 2的幂
 */

// @lc code=start
class Solution
{
public:
    bool isPowerOfTwo(int n)
    {
        return n > 0 && !(n & (n - 1));
    }
};
// @lc code=end

File: rust-answers/231.2-的幂.rs
/*
 * @lc app=leetcode.cn id=231 lang=rust
 *
 * [231] 2 的幂
 */

// @lc code=start
impl Solution {
    pub fn is_power_of_two(n: i32) -> bool {
        // n > 0 && n.count_ones() == 1
        n > 0 && (n & (n - 1)) == 0
    }
}
// @lc code=end

